ddp-workspace/projects/ddp-singular/src/app/components/header/header.component.ts

```diff
@@ -48,4 +51,13 @@ export class HeaderComponent {
       this.isNavigationShown = false;
     }
   }
+
+  signIn(): void {
+    this.analytics.emitCustomGtagEvent(GTagEvent.LOG_IN_CLICK);
+  }
+
+  signUp(event: Event): void {
+    const link = event.currentTarget as HTMLAnchorElement;
+    this.analytics.emitCustomGtagEvent(GTagEvent.SIGN_UP_CLICK, link.innerText, link.href);
```

Comment:
As an option (in order to avoid lots of duplication) we can extract all gtag related stuff to a parent class. 
It will be responsible for emiting **gtag** events only.
Other components can extend that parent class and use a common method to send analytics.

Reply:
I see. Makes sense. In this case, though there are very specific events being requested and using gtag.
It is likely that we will eventually move EVERYTHING ELSE to either gtag.js or a Analytics 4. I don't know. But will happen in the next couple of months.
This particular PR is to satisfy some very specific requests without breaking existing functionality. We will get opportunity to refactor then. For now, there is some hurry to get it in!